tools/tests/add_and_update.py

```diff
@@ -0,0 +1,136 @@
+#! /usr/bin/env python3
+
+""" Test `antler-proj add` and `antler proj update` commands. """
+
+import os
+import shutil
+
+from util import antler_in_proj_cmd, load_project
+from init_tests import init_project
+
+PROJECT_NAME="aou_project"
+PROJECT_PATH=os.path.join("./",PROJECT_NAME)
+
+def expect_subvalue(array_of_lists, name, tag, value):
+    """ Inside array_of_lists, find the list that contains `name` and test the value
+        stored in that list for the given tag against value.
+
+
+    :param array_of_lists  An array of lists.
+    :param name  The name to search for.
+    :tag  The tag that contains the value we to test.
+    :value  The expected value.
+
+    """
+
+    for i in array_of_lists:
+        if i["name"] == name:
+            return i[tag] == value
+    return False
+
+
+def setup_project():
+    """Setup the project ensuring there is a fresh project file.
+    """
+
+    shutil.rmtree(PROJECT_PATH, ignore_errors=True)
+    init_project(PROJECT_NAME, "v1.0.0", PROJECT_PATH)
+    assert os.path.isdir(PROJECT_PATH)
+
+
+def test_app_compile_options():
+    """Add an application to the project and test its compile options can be updated.
+    """
+
+    out, _ = antler_in_proj_cmd(PROJECT_PATH, "add app App1 C++ \\\\-O1")
```

Comment:
I'm trying to understand how `\\\\-o1` turns into `-o1`.
So your transform function should transform `\\\\-o1` -> `\\-o1`. what removes the rest?

Reply:
The shell. :-)